core/execution/market.go

```diff
@@ -774,6 +774,47 @@ func (m *Market) updateMarketValueProxy() {
 	m.stateChanged = true
 }
 
+func (m *Market) closeCancelledMarket(ctx context.Context, t time.Time) error {
```

Comment:
this whole function is identical to close market from line 800 onwards. I'd suggest refactoring closeMarket to split out the settlement part rather than repeating 40 lines here.

Reply:
The refactoring is done here https://github.com/vegaprotocol/vega/pull/5668/commits/a11e922add05a5659da732ef070f6e0cc73cb00f#diff-1782f5982916957ff0e7e82b0c7130c6d4c8ab4eb2e06f5c7b5b4a5925e15bcfR777
Thanks!